IHP/AutoRefresh.hs

```diff
@@ -213,7 +220,26 @@ gcSessions autoRefreshServer = do
 isSessionExpired :: UTCTime -> AutoRefreshSession -> Bool
 isSessionExpired now AutoRefreshSession { lastPing } = (now `diffUTCTime` lastPing) > (secondsToNominalDiffTime 60)
 
--- | Stops all async Auto Refresh subscriptions
-stopAutoRefreshServer :: IORef AutoRefreshServer -> IO ()
-stopAutoRefreshServer autoRefreshServer =
-    readIORef autoRefreshServer >>= (\autoRefreshServer -> autoRefreshServer |> get #subscriptions |> mapM_ uninterruptibleCancel)
+-- | Returns the event name of the event that the pg notify trigger dispatches
+channelName :: ByteString -> ByteString
+channelName tableName = "did_change_" <> tableName
+
+-- | Returns the sql code to set up a database trigger
+notificationTrigger :: ByteString -> PG.Query
+notificationTrigger tableName = PG.Query $ ""
+        <> "BEGIN;\n"
+        <> "CREATE OR REPLACE FUNCTION " <> functionName <> "() RETURNS TRIGGER AS $$"
```

Comment:
Maybe use `[trimmed||]` instead of string concatenation? Could make it a bit more readable

Reply:
Good point 👍  will apply this